components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java

```diff
@@ -1327,11 +1348,35 @@ public UserStoreAttributeMapping getUserStoreMappingAttributes(String typeId) {
                     getUserStoreAttributeMappings();
             Map<String, UserStoreAttributeDO> mapping = userStoreAttributeMappings
                     .getUserStoreAttributeMappings(userStoreName);
+
+            // Remove identity claim mappings by iterating through all the claim mappings.
+            if (excludeIdentityClaimMappings) {
+                return excludeIdentityClaims(mapping);
+            }
             return new ArrayList<>(mapping.values());
         } catch (IdentityUserStoreMgtException e) {
             LOG.error("Error occurred while retrieving user store attribute metadata", e);
             throw handleException(Response.Status.INTERNAL_SERVER_ERROR, UserStoreConstants.ErrorMessage.
                     ERROR_CODE_ERROR_RETRIEVING_USER_STORE_ATTRIBUTE_METADATA);
         }
     }
+
+    /**
+     * Exclude the userstore attributes with identity claims.
+     *
+     * @param userStoreAttributeDOMap Userstore attribute map.
+     * @return List of UserStoreAttributeDOs.
+     */
+    private List<UserStoreAttributeDO> excludeIdentityClaims(Map<String,
+            UserStoreAttributeDO> userStoreAttributeDOMap) {
```

Comment:
formatting issue
```suggestion
    private List<UserStoreAttributeDO> excludeIdentityClaims(
            Map<String, UserStoreAttributeDO> userStoreAttributeDOMap) {
```

Reply:
Missed this comment. Addressed this with https://github.com/wso2/identity-api-server/pull/318